crates/hotshot/src/traits/election/static_committee_leader_two_views.rs

```diff
@@ -0,0 +1,217 @@
+use std::{marker::PhantomData, num::NonZeroU64};
+
+use ethereum_types::U256;
+// use ark_bls12_381::Parameters as Param381;
+use hotshot_types::traits::signature_key::StakeTableEntryType;
+use hotshot_types::{
+    signature_key::BLSPubKey,
+    traits::{election::Membership, node_implementation::NodeType, signature_key::SignatureKey},
+    PeerConfig,
+};
+#[cfg(feature = "randomized-leader-election")]
+use rand::{rngs::StdRng, Rng};
+use tracing::debug;
+
+/// Dummy implementation of [`Membership`]
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct StaticCommitteeLeaderForTwoViews<T, PUBKEY: SignatureKey> {
+    /// All the nodes participating and their stake
+    all_nodes_with_stake: Vec<PUBKEY::StakeTableEntry>,
+    /// The nodes on the static committee and their stake
+    committee_nodes_with_stake: Vec<PUBKEY::StakeTableEntry>,
+    /// builder nodes
+    committee_nodes_without_stake: Vec<PUBKEY>,
+    /// the number of fixed leader for gpuvid
+    fixed_leader_for_gpuvid: usize,
+    /// Node type phantom
+    _type_phantom: PhantomData<T>,
+}
+
+/// static committee using a vrf kp
+pub type StaticCommittee<T> = StaticCommitteeLeaderForTwoViews<T, BLSPubKey>;
+
+impl<T, PUBKEY: SignatureKey> StaticCommitteeLeaderForTwoViews<T, PUBKEY> {
+    /// Creates a new dummy elector
+    #[must_use]
+    pub fn new(
+        _nodes: &[PUBKEY],
+        nodes_with_stake: Vec<PUBKEY::StakeTableEntry>,
+        nodes_without_stake: Vec<PUBKEY>,
+        fixed_leader_for_gpuvid: usize,
+    ) -> Self {
+        Self {
+            all_nodes_with_stake: nodes_with_stake.clone(),
+            committee_nodes_with_stake: nodes_with_stake,
+            committee_nodes_without_stake: nodes_without_stake,
+            fixed_leader_for_gpuvid,
+            _type_phantom: PhantomData,
+        }
+    }
+}
+
+impl<TYPES, PUBKEY: SignatureKey + 'static> Membership<TYPES>
+    for StaticCommitteeLeaderForTwoViews<TYPES, PUBKEY>
+where
+    TYPES: NodeType<SignatureKey = PUBKEY>,
+{
+    /// Clone the public key and corresponding stake table for current elected committee
+    fn committee_qc_stake_table(&self) -> Vec<PUBKEY::StakeTableEntry> {
+        self.committee_nodes_with_stake.clone()
+    }
+
+    #[cfg(not(any(
+        feature = "randomized-leader-election",
+        feature = "fixed-leader-election"
+    )))]
+    /// Index the vector of public keys with the current view number
+    fn leader(&self, view_number: TYPES::Time) -> PUBKEY {
+        // two connsecutive views will have same index starting with even number.
+        // eg 0->1, 2->3 ... 10->11 etc
+        let index =
+            usize::try_from((*view_number / 2) % self.all_nodes_with_stake.len() as u64).unwrap();
+        let res = self.all_nodes_with_stake[index].clone();
+        TYPES::SignatureKey::public_key(&res)
+    }
+
+    #[cfg(feature = "fixed-leader-election")]
```

Comment:
See above comment, but given this impl is specifically for a certain leader election we should not be implementing the fixed leader version